### PR TITLE
Support other promise implementations

### PR DESCRIPTION
**Pull request by ixalon**
But leave Q as the default for backwards compatibility.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,10 +24,12 @@
     "combine"
   ],
   "author": "Beamly ltd.",
+  "license": "UNLICENSED",
```
Comment (jonathan-fielding) [inline package.json]: why is it unlicensed? this is an open source project

File: src/object.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-var Q = require('q');
+var promise = require('./promise');
```
Comment (jonathan-fielding) [inline src/object.js]: could this be named something different, it feels like Promise is a key word so shouldn't be a variable you use, perhaps promiseImp or even promiseLib

File: package.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
 {
   "name": "q-combinators",
-  "version": "0.6.1",
+  "version": "0.7.0",
```
Comment (jonathan-fielding) [inline package.json]: could this be our 1.0.0 release?

File: package.json [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
   "description": "Functions to combine q promises, capturing lots of useful, real world patterns.",
   "main": "index.js",
   "dependencies": {
-    "q": "1.0.0",
-    "lodash": "2.4.1"
+    "lodash": "2.4.1",
```
Comment (jonathan-fielding) [inline package.json]: could we update lodash, this is a very old version and i beleive it throws warnigns on install

File: package.json (hunk 1 above)
Comment (ixalon) [inline package.json]: Because master has no license specified. A correct license should be given to this module in a separate pull request.

File: package.json (hunk 4 above)
Comment (ixalon) [inline package.json]: There is a separate PR for that: https://github.com/beamly/q-combinators/pull/17

File: package.json (hunk 3 above)
Comment (ixalon) [inline package.json]: Suggest moving to 1.0.0 once https://github.com/beamly/q-combinators/pull/17 is merged.

**Comment by jonathan-fielding** [comment]
👍
